WooCommerce/Classes/ViewRelated/Products/Variations/Add Attributes/AddAttributeOptionsViewModel.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import Yosemite
+
+/// Provides view data for Add Attributes, and handles init/UI/navigation actions needed.
+///
+final class AddAttributeOptionsViewModel {
+
+    typealias Section = AddAttributeOptionsViewController.Section
+    typealias Row = AddAttributeOptionsViewController.Row
+
+    private(set) var newAttributeName: String?
+    private(set) var attribute: ProductAttribute?
+
+    private(set) var sections: [Section] = []
+
+    init(newAttribute: String?) {
+        self.newAttributeName = newAttribute
+        updateSections()
+    }
+
+    init(existingAttribute: ProductAttribute) {
+        self.attribute = existingAttribute
+        updateSections()
+    }
+
+}
+
+// MARK: - Synchronize Product Attribute terms
+//
+private extension AddAttributeOptionsViewModel {
+    // TODO: to be implemented - fetch of terms
+
+    /// Updates  data in sections
+    ///
+    func updateSections() {
+        let textFieldSection = Section(header: nil, footer: Localization.footerTextField, rows: [.termTextField])
+        let selectedTermsSection = Section(header: Localization.headerSelectedTerms, footer: nil, rows: [.selectedTerms])
+        let existingTermsSection = Section(header: Localization.headerExistingTerms, footer: nil, rows: [.existingTerms])
+        sections = [textFieldSection, selectedTermsSection, existingTermsSection].compactMap { $0 }
```

Comment:
nit: is `compactMap` necessary?

Reply:
Currently no, because the sections are static, but in the future yes, since the selectedTermsSection can contain zero elements, so it should be hidden 👍